### PR TITLE
refactor(FX-4682): Move save artwork button to artwork actions

### PR DESCRIPTION
**Pull request by gkartalis**
This PR resolves [FX-4682] 

### Description

After the redesign of the Artwork Page we noticed a significant drop on artwork saves, so we decided to move it where it was placed before (under the artwork image -> ArtworkActions component)

This pr includes the following:
- [x] refactored artwork action screen -> functional component
- [x] moves save button to artworkActions (below artwork image)
- [x] update tests to reflect the changes
- [x] update the artwork screen placeholders to reflect the changes
- [x] **bonus** increased the hitslop of the artwork action buttons to make them more easily pressable
- [x] cleanup dead code from artworkscreenheader 
  

| Before | After |
|---|---|
|Simulator Screen Shot - iPhone 13 - 2023-03-15 at 16 19 01|Simulator Screen Shot - iPhone 13 - 2023-03-15 at 16 38 48|
|IMG_C2A093B6FDA1-1|Simulator Screen Shot - iPhone 13 - 2023-03-15 at 16 51 37|
|Simulator Screen Shot - iPhone 13 - 2023-03-15 at 17 06 03|Simulator Screen Shot - iPhone 13 - 2023-03-15 at 16 53 53|

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Artwork Screen moved save artwork button to artwork actions - gkartalis

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[FX-4682]: https://artsyproduct.atlassian.net/browse/FX-4682?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Scenes/Artwork/Components/AboveTheFoldArtworkPlaceholder.tsx [hunk 1]
```diff
@@ -65,7 +64,6 @@ export const AboveTheFoldPlaceholder: React.FC<AboveTheFoldPlaceholderProps> = (
           <PlaceholderBox width={20} height={20} />
 
           <Flex flexDirection="row" alignItems="center">
-            <PlaceholderBox width={105} height={25} marginRight={space(1)} />
```
Comment (gkartalis) [inline src/app/Scenes/Artwork/Components/AboveTheFoldArtworkPlaceholder.tsx:68]: removes the placeholder box for the header save icon (since we removed it from there)

File: src/app/Scenes/Artwork/Components/AboveTheFoldArtworkPlaceholder.tsx [hunk 2]
```diff
@@ -14,7 +14,7 @@ const ArtworkActionsPlaceholder = () => {
 
   return (
     <Flex flexDirection="row" justifyContent="center">
-      {times(2).map((index) => (
+      {times(3).map((index) => (
```
Comment (gkartalis) [inline src/app/Scenes/Artwork/Components/AboveTheFoldArtworkPlaceholder.tsx:17]: updates artwork actions placeholder to have 3 buttons (save / share / view in room)

File: src/app/Scenes/Artwork/Components/ArtworkActions.tests.tsx [hunk 3]
```diff
@@ -112,21 +120,66 @@ describe("ArtworkActions", () => {
...
+        <ArtworkActionsFragmentContainer artwork={artwork!} shareOnPress={() => jest.fn()} />
+      ),
+      query: graphql`
+        query ArtworkActionsTestQuery @relay_test_operation @raw_response_type {
+          artwork(id: "some-artwork") {
+            ...ArtworkActions_artwork
+          }
+        }
+      `,
+    })
```
Comment (gkartalis) [inline src/app/Scenes/Artwork/Components/ArtworkActions.tests.tsx:136]: for now I added the `setupTestWrapper` only on this test but will pull it out on another pr for all the tests of this file to use that

File: src/app/Scenes/Artwork/Components/ArtworkActions.tsx [hunk 4]
```diff
@@ -97,13 +186,10 @@ export const ArtworkActionsFragmentContainer = createFragmentContainer(ArtworkAc
...
     fragment ArtworkActions_artwork on Artwork {
       id
+      internalID
       slug
-      title
-      href
       isHangable
-      artists {
-        name
-      }
```
Comment (dimatretyak) [inline src/app/Scenes/Artwork/Components/ArtworkActions.tsx:106]: **praise**: 👌👌👌

File: src/app/Scenes/Artwork/Components/ArtworkActions.tsx [hunk 5]
```diff
@@ -61,30 +92,88 @@ export class ArtworkActions extends React.Component<ArtworkActionsProps> {
...
+      {!!(LegacyNativeModules.ARCocoaConstantsModule.AREnabled && isHangable) && (
+        <Touchable
+          hitSlop={{
+            top: space(1),
+            bottom: space(1),
+          }}
+          haptic
+          onPress={() => openViewInRoom()}
+        >
+          <UtilButton pr={2}>
```
Comment (dimatretyak) [inline src/app/Scenes/Artwork/Components/ArtworkActions.tsx:155]: **suggestion**: replace `pr={2}` for `UtilButton` on `Join`. Something like this

```javascript
<Join separator={<Spacer x={2} />}>
  {/* Save Button */}
  <Touchable>...</Touchable>

  {/* View in Room */}
  <Touchable>...</Touchable>

  {/* Share */}
  <Touchable>...</Touchable>
</Join>
```

File: src/app/Scenes/Artwork/Components/ArtworkActions.tsx [hunk 6]
```diff
@@ -61,30 +92,88 @@ export class ArtworkActions extends React.Component<ArtworkActionsProps> {
...
+        accessibilityRole="button"
+        accessibilityLabel="Save artwork"
+        onPress={handleArtworkSave}
+      >
+        {/* <Flex flex={1} justifyContent="center" alignItems="center" flexDirection="row" pr={2}> */}
+        <UtilButton pr={2}>
+          <SaveIcon isSaved={!!isSaved} />
+          <Spacer x={0.5} />
+          {/* the spaces below are to not make the icon jumpy when changing from save to saved will work on a more permanent fix */}
+          <Text variant="sm">{isSaved ? "Saved" : "Save   "}</Text>
```
[the remark this answers is not in the document]
Comment (dimatretyak) [reply src/app/Scenes/Artwork/Components/ArtworkActions.tsx:142]: and should we display "Watch Lot" label when artwork is in auction ([how it was before](https://github.com/artsy/eigen/blob/daa98c1a5171bab2a2ee3e2549a0b366036adee3/src/app/Scenes/Artwork/Components/ArtworkActions.tsx#L142)).

[the remark this answers is not in the document]
Comment (gkartalis) [reply src/app/Scenes/Artwork/Components/ArtworkActions.tsx:142]: I thought we went with using just `Save/Saved` but will open a thread about that 👍

File: src/app/Scenes/Artwork/Components/ArtworkActions.tsx [hunk 7]
```diff
@@ -61,30 +92,88 @@ export class ArtworkActions extends React.Component<ArtworkActionsProps> {
...
+        hitSlop={{
+          top: space(1),
+          left: space(1),
+          bottom: space(1),
+        }}
+        accessibilityRole="button"
+        accessibilityLabel="Save artwork"
+        onPress={handleArtworkSave}
+      >
+        {/* <Flex flex={1} justifyContent="center" alignItems="center" flexDirection="row" pr={2}> */}
```
[the remark this answers is not in the document]
Comment (gkartalis) [reply src/app/Scenes/Artwork/Components/ArtworkActions.tsx:137]: 🙈

File: src/app/Scenes/Artwork/Components/ArtworkActions.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (dimatretyak) [reply src/app/Scenes/Artwork/Components/ArtworkActions.tsx:142]: thanks!

File: src/app/Scenes/Artwork/Components/ArtworkActions.tsx [hunk 8]
```diff
@@ -61,30 +92,88 @@ export class ArtworkActions extends React.Component<ArtworkActionsProps> {
...
+        <UtilButton pr={2}>
+          <SaveIcon isSaved={!!isSaved} />
+          <Spacer x={0.5} />
+          {/* the spaces below are to not make the icon jumpy when changing from save to saved will work on a more permanent fix */}
+          <Text variant="sm">{isSaved ? "Saved" : "Save   "}</Text>
+          {/* </Flex> */}
+        </UtilButton>
+      </Touchable>
```
[the remark this answers is not in the document]
Comment (gkartalis) [reply src/app/Scenes/Artwork/Components/ArtworkActions.tsx:145]: Will do, although at some point we really need to rethink the folder structure of the Artwork Screen in eigen it is so messy 🤔 so many components in one folder.

But I will move this and follow up on the structure on another pr

**Review by anandaroop — APPROVED** [review]
